File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/xmi/SerializationContext.java
package org.kevoree.modeling.api.xmi;

import org.kevoree.modeling.api.Callback;
import org.kevoree.modeling.api.KConfig;
import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.api.KModelAttributeVisitor;
import org.kevoree.modeling.api.map.LongHashMap;
import org.kevoree.modeling.api.map.StringHashMap;

import java.util.ArrayList;

class SerializationContext {
    public boolean ignoreGeneratedID = false;
    public KObject model;
    public Callback<String> finishCallback;
    public StringBuilder printer;
    public KModelAttributeVisitor attributesVisitor;

    // KPath -> XMIPath
    LongHashMap<String> addressTable = new LongHashMap<String>(KConfig.CACHE_INIT_SIZE, KConfig.CACHE_LOAD_FACTOR);
    // KPath -> Count
    StringHashMap<Integer> elementsCount = new StringHashMap<Integer>(KConfig.CACHE_INIT_SIZE, KConfig.CACHE_LOAD_FACTOR);

    ArrayList<String> packageList = new ArrayList<String>();

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/xmi/SerializationContext.ts
///<reference path="../KObject.ts"/>
///<reference path="../ThrowableCallback.ts"/>
///<reference path="../../../../../java/util/JUArrayList.ts"/>
///<reference path="../../../../../java/util/JUHashMap.ts"/>

class SerializationContext {

  public ignoreGeneratedID: boolean = false;
  public model: KObject<any,any> = null;
  public finishCallback: ThrowableCallback<string> = null;
  public printer: StringBuilder = null;
  public attributesVisitor: AttributesVisitor = null;
  public addressTable: JUHashMap<number, string> = new JUHashMap<number, string>();
  public elementsCount: JUHashMap<string, number> = new JUHashMap<string, number>();
  public packageList: JUArrayList<string> = new JUArrayList<string>();

}

